tests/test_start_date.py

```diff
@@ -0,0 +1,125 @@
+
+        expected_streams = self.expected_streams()
+
+        conn_id_1 = connections.ensure_connection(self, original_properties=False)
+        runner.run_check_mode(self, conn_id_1)
+
+        found_catalogs_1 = self.run_and_verify_check_mode(conn_id_1)
+        self.select_found_catalogs(conn_id_1, found_catalogs_1, only_streams=expected_streams)
+
+        sync_record_count_1 = self.run_and_verify_sync(conn_id_1)
+
+        synced_records_1 = runner.get_records_from_target_output()
+
+        ##########################################################################
+        ### Update Start Date for 2nd sync
+        ##########################################################################
+
+        self.START_DATE = self.second_start_date
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        conn_id_2 = connections.ensure_connection(self, original_properties=False)
+        runner.run_check_mode(self, conn_id_2)
+
+        found_catalogs_2 = self.run_and_verify_check_mode(conn_id_2)
+        self.select_found_catalogs(conn_id_2, found_catalogs_2, only_streams=expected_streams)
+
+        sync_record_count_2 = self.run_and_verify_sync(conn_id_2)
+
+        synced_records_2 = runner.get_records_from_target_output()
+
+        self.assertGreaterEqual(sum(sync_record_count_1.values()), sum(sync_record_count_2.values()))
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_replication_keys = self.expected_replication_keys()[stream]
+                expected_metadata = self.expected_metadata()[stream]
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                record_count_sync_1 = sync_record_count_1.get(stream, 0)
+                record_count_sync_2 = sync_record_count_2.get(stream, 0)
+                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_1.get(stream, {}).get('messages')
+                                       if message.get('action') == 'upsert']
+                primary_keys_list_2 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                                       for message in synced_records_2.get(stream, {}).get('messages')
+                                       if message.get('action') == 'upsert']
+
+                primary_keys_sync_1 = set(primary_keys_list_1)
+                primary_keys_sync_2 = set(primary_keys_list_2)
+
+                if expected_metadata.get(self.OBEYS_START_DATE):
```

Comment:
This should be a hard get. As in `expected_metadata[self.OBEYS_START_DATE]` over `expected_metadata.get(self.OBEYS_START_DATE)`, so that if an expectation is not set the test will throw a KeyError.

Reply:
Updated the code to use hard get.